python/python/lance/__init__.py

```diff
@@ -37,6 +30,9 @@
 from .schema import json_to_schema, schema_to_json
 from .util import sanitize_ts
 
+ts_types = Union[datetime, pd.Timestamp, str]
```

Comment:
Will `pd.Timestamp` error here if pandas is not available?

Reply:
Yes they do. I propose moving all the type definitions into `if TYPE_CHECKING` blocks.